sys/mips/timer.c

```diff
@@ -27,9 +29,22 @@ static bintime_t mips_timer_gettime(timer_t *tm);
 static uint64_t read_count(mips_timer_state_t *state) {
   SCOPED_INTR_DISABLED();
   state->count.lo = mips32_get_c0(C0_COUNT);
+
   /* detect hardware counter overflow */
-  if (state->count.lo < state->last_count_lo)
+  if (state->count.lo < state->last_count_lo) {
+    state->cntr_modulo +=
+      ((uint32_t)(-1) - state->last_count_lo) + state->count.lo;
     state->count.hi++;
+  } else {
+    state->cntr_modulo += state->count.lo - state->last_count_lo;
+  }
+
+  if (state->cntr_modulo >= state->timer.tm_frequency) {
```

Comment:
could you please explain or comment this line?

Reply:
The `cntr_modulo` shows how many ticks elapsed since timer initialization modulo `tm_frequency`, at each time update we increase `cntr_modulo` with number of ticks which elapsed since the last check and if the value reaches `tm_frequency` it means that one second passed and we want to 'normalize' the counter.